x/ecocredit/core/genesis_test.go

```diff
@@ -0,0 +1,234 @@
+package core_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/regen-network/regen-ledger/x/ecocredit/core"
+)
+
+func TestValidateGenesis(t *testing.T) {
+	x := `{
+		"regen.ecocredit.v1.BatchBalance":[
+			{"address":"gydQIvR2RUi0N1RJnmgOLVSkcd4=","batch_id":"1","tradable":"90.003","retired":"9.997","escrowed":""}
+		],
+		"regen.ecocredit.v1.BatchInfo":[
+			{"issuer":"WCBEyNFP/N5RoS4h43AqkjC6zA8=","project_id":"1","batch_denom":"BIO01-00000000-00000000-001","metadata":"batch metadata","start_date":null,"end_date":null,"issuance_date":"2022-04-08T10:40:10.774108141Z"},
```

Comment:
Any reason for using `null` for some of these values and not for others? Might be nice to only include the values necessary for validation or to include values for all properties. For example, `start_date` is `null` but `issuance_date` is not and neither are necessary for `GenesisValidate` to pass.

Reply:
> start_date is null but issuance_date is not and neither are necessary for GenesisValidate to pass.

Yes. Those fields are not required for genesis validation. I'll remove them.

I'm also thinking about adding validation for each message type to check all fields are properly set.

Reply:
That sounds like the right direction to me.